path_demo.cpp

```diff
@@ -25,17 +25,41 @@ int main(int argc, char **argv) {
     cout << "some/path.ext:operator==() (unequal) = " << (path("some/path.ext") == path("another/path.ext")) << endl;
 
     cout << "nonexistant:exists = " << path("nonexistant").exists() << endl;
+    try
+    {
+        cout << "nonexistant:file_size = " << path("nonexistant").file_size() << endl;
+    }
+    catch(std::runtime_error e)
+    {
+        cout << "<file does not exist>" << endl;
+    }
     cout << "nonexistant:is_file = " << path("nonexistant").is_file() << endl;
     cout << "nonexistant:is_directory = " << path("nonexistant").is_directory() << endl;
     cout << "nonexistant:filename = " << path("nonexistant").filename() << endl;
     cout << "nonexistant:extension = " << path("nonexistant").extension() << endl;
+    try
+    {
+        cout << "filesystem/path.h:file_size = " << path("filesystem/path.h").file_size() << endl;
+    }
+    catch(std::runtime_error e)
+    {
+        cout << "<file does not exist>" << endl;
+    }
     cout << "filesystem/path.h:exists = " << path("filesystem/path.h").exists() << endl;
     cout << "filesystem/path.h:is_file = " << path("filesystem/path.h").is_file() << endl;
     cout << "filesystem/path.h:is_directory = " << path("filesystem/path.h").is_directory() << endl;
     cout << "filesystem/path.h:filename = " << path("filesystem/path.h").filename() << endl;
     cout << "filesystem/path.h:extension = " << path("filesystem/path.h").extension() << endl;
     cout << "filesystem/path.h:make_absolute = " << path("filesystem/path.h").make_absolute() << endl;
     cout << "../filesystem:exists = " << path("../filesystem").exists() << endl;
+    try
+    {
+        cout << "../filesystem:file_size = " << path("../filesystem").file_size() << endl;
+    }
+    catch(std::runtime_error e)
+    {
+        cout << "<file does not exist>" << endl;
```

Comment:
this part of the try block actually gets executed
output looks like you would expect:

    nonexistant:file_size = <file does not exist>